tools/filesystem/filesystem.go

```diff
@@ -113,23 +113,20 @@ func (s *System) Delete(fileKey string) error {
 
 // DeletePrefix deletes everything starting with the specified prefix.
 func (s *System) DeletePrefix(prefix string) []error {
-	failed := []error{}
-
 	if prefix == "" {
-		failed = append(failed, errors.New("Prefix mustn't be empty."))
-		return failed
+		return []error{errors.New("Prefix mustn't be empty.")}
 	}
 
-	dirsMap := map[string]struct{}{}
-	dirsMap[prefix] = struct{}{}
-
-	opts := blob.ListOptions{
-		Prefix: prefix,
+	failed := make([]error, 0)
```

Comment:
Since we don't have a cap, let's use the literal (eg. `failed := []error{}`) in this case for consistency with the dirsMap.